test/jdk/sun/security/tools/jarsigner/CheckSignerCertChain.java

```diff
@@ -88,5 +88,32 @@ public static void main(String[] args) throws Exception {
                 // key, but not for its SHA1withRSA algorithm.
                 .shouldContain("Signature algorithm: SHA1withRSA, 1024-bit key (weak)")
                 .shouldHaveExitValue(0);
```

Comment:
What does the test show before this fix? If `jdk.jar.dA` and `jdk.certpath.dA` are the same, then there's no way to find out if the new code works.

Reply:
Added test using new java.security with different disabledAlgorithms for certpath and jar.